### PR TITLE
DESENG-493: CRON config alignment and engagement metadata dependency removal

### PR DESCRIPTION
**Pull request by VineetBala-AOT**
Issue #: https://apps.itsm.gov.bc.ca/jira/browse/DESENG-493

*Description of changes:*
- Aligned the CRON configuration and sample environment files with the structure used in the Met API.
- Eliminated the reliance on engagement metadata within cron jobs, improving their independence.
- Implemented necessary code adjustments to seamlessly integrate with the updated CRON configuration.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-api/src/met_api/services/engagement_service.py [hunk 1]
```diff
@@ -295,7 +295,7 @@ def _render_email_template(engagement: EngagementModel):
         engagement_url = notification.get_tenant_site_url(engagement.tenant_id, dashboard_path)
         templates = current_app.config['EMAIL_TEMPLATES']
         subject = templates['CLOSEOUT']['SUBJECT'].format(engagement_name=engagement.name)
-        email_environment = templates['ENVIROMENT']
+        email_environment = templates['ENVIRONMENT']
```
Comment (Baelx) [inline met-api/src/met_api/services/engagement_service.py:298]: Good catch!

File: met-cron/sample.env [hunk 2]
```diff
@@ -1,20 +1,96 @@
...
-MET_DATABASE_PASSWORD=met
-MET_DB_NAME=met
-MET_DATABASE_HOST=localhost
-MET_DATABASE_PORT=5432
+USE_DEBUG=True # Enable a dev-friendly debug mode
+TESTING= # Handle errors normally (False) or raise exceptions (True)
 
+# Miscellaneous Settings
+SECRET_KEY="" # For Flask sessions. If unset, this value is randomized
+# disables certain checks for user permissions and tenant access. Buggy.
```
Comment (Baelx) [inline met-cron/sample.env:13]: Does this comment mean that when you disable certain checks for user permissions and tenant access that unexpected behaviour may occur? Why would a developer want to disable those checks in the first place?

File: met-cron/sample.env [hunk 3]
```diff
@@ -1,20 +1,96 @@
...
+KEYCLOAK_CONNECT_TIMEOUT="60"
+
+# JWT OIDC configuration for authentication
+# Populate from 'GDX MET web (public)-installation-*.json'
+JWT_OIDC_AUDIENCE="" # resource
+JWT_OIDC_ISSUER="" # default: constructed from base url and realm name
+JWT_OIDC_WELL_KNOWN_CONFIG="" # default: constructed from issuer
+JWT_OIDC_JWKS_URI=""          # default: constructed from issuer
+# Object path to access roles from JWT token
+# JWT_OIDC_ROLE_CLAIM=realm_access.roles # SSO schema
```
Comment (Baelx) [inline met-cron/sample.env:35]: We should remove this commented-out line if it's not being used

File: met-cron/sample.env [hunk 4]
```diff
@@ -1,20 +1,96 @@
...
+SQLALCHEMY_ECHO= 
+SQLALCHEMY_TRACK_MODIFICATIONS=
+
+# Email API Configuration
+NOTIFICATIONS_EMAIL_ENDPOINT=https://met-notify-api-dev.apps.gold.devops.gov.bc.ca/api/v1/notifications/email
+EMAIL_SECRET_KEY="notASecureKey" # If unset, this value is randomized
+EMAIL_ENVIRONMENT=
+EMAIL_FROM_ADDRESS="met-example@gov.bc.ca"
+# Email Template Configuration
+# Subject lines have a reasonable default value
```
Comment (Baelx) [inline met-cron/sample.env:57]: Could you explain this comment a little bit more please?

File: met-cron/sample.env [hunk 5]
```diff
@@ -1,20 +1,96 @@
...
+
+# Test database settings
+# If unset, uses the same settings as the main database
+DATABASE_TEST_USERNAME=
+DATABASE_TEST_PASSWORD=
+DATABASE_TEST_NAME=
+DATABASE_TEST_HOST=
+DATABASE_TEST_PORT=
+
+# A keycloak server is started automatically by Pytest; there is no need to start your own instance.
```
Comment (Baelx) [inline met-cron/sample.env:87]: Thanks for this comment!

File: met-cron/src/met_cron/services/mail_service.py [hunk 6]
```diff
@@ -53,17 +53,19 @@ def _send_email_notification_for_subscription(engagement_id, template_id, subjec
     def _render_email_template(engagement, participant, template):
         site_url = notification.get_tenant_site_url(engagement.tenant_id)
         tenant_name = EmailService._get_tenant_name(engagement.tenant_id)
-        metadata_model: EngagementMetadataModel = EngagementMetadataModel.find_by_id(engagement.id)
-        project_name = None
-        if metadata_model and 'project_name' in metadata_model.project_metadata:
-            project_name = metadata_model.project_metadata.get('project_name')
+        # TODO should be re-visited once the engagement metadata functionality of completed
```
Comment (Baelx) [inline met-cron/src/met_cron/services/mail_service.py:56]: Thanks for the TODO comments. This is exactly what we want to see when we're commenting out code

**Review by Baelx — CHANGES_REQUESTED** [review]
Nice work! 

Could you please just remove a commented-out environment variable that appears to be duplicated. Also, I feel some comments could be a bit clearer for developers. Could you please respond to my questions about certain comments?

File: met-cron/sample.env (hunk 2 above)
Comment (VineetBala-AOT) [reply met-cron/sample.env]: I removed this comment as it's not applicable to our scenario. The checks in this case are associated with the tenant. In a multi-tenant environment like ours, we verify if this variable is set to false. In such cases, we interpret the application as multi-tenant, and consequently, we assess user access to a specific tenant to restrict engagement access within that tenant. Conversely, if this variable is set to true, we consider it a single-tenant environment, eliminating the need to check user tenant access.

File: met-cron/sample.env (hunk 3 above)
Comment (VineetBala-AOT) [reply met-cron/sample.env]: Removed it, thanks.

File: met-cron/sample.env (hunk 4 above)
Comment (VineetBala-AOT) [reply met-cron/sample.env]: I revised the comment to enhance clarity.

**Review by Baelx — APPROVED** [review]

**Pull request merged**